polkadot/runtime/parachains/src/util.rs

```diff
@@ -98,13 +104,104 @@ pub fn take_active_subset<T: Clone>(active: &[ValidatorIndex], set: &[T]) -> Vec
 	subset
 }
 
+/// On first pass it filters out all candidates that have multiple cores assigned and no
+/// `CoreIndex` injected.
+///
+/// On second pass we filter out candidates with the same core index. This can happen if for example
+/// collators distribute collations to multiple backing groups.
+pub(crate) fn filter_elastic_scaling_candidates<T: configuration::Config + scheduler::Config>(
+	candidates: &mut Vec<BackedCandidate<T::Hash>>,
+) {
+	if !configuration::Pallet::<T>::config()
+		.node_features
+		.get(FeatureIndex::ElasticScalingCoreIndex as usize)
+		.map(|b| *b)
+		.unwrap_or(false)
+	{
+		// we don't touch the candidates, since we don't expect block producers
+		// to inject `CoreIndex`.
+		return
+	}
+
+	// A mapping from parachain to all assigned cores.
+	let mut cores_per_parachain: BTreeMap<ParaId, Vec<CoreIndex>> = BTreeMap::new();
+
+	for (core_index, para_id) in <scheduler::Pallet<T>>::scheduled_paras() {
+		cores_per_parachain.entry(para_id).or_default().push(core_index);
+	}
+
+	// We keep a candidate if the parachain has only one core assigned or if
```

Comment:
shouldn't we do the same thing if ElasticScalingMVP feature is disabled? only keep one candidate

Reply:
I added this